### DISCUSSION TITLE
Typescript issue when using createStore + persist

### DISCUSSION THREAD

**Discussion by De-Santa** [pmndrs/zustand#2068] [Q&A] [answered]
## Summary
Issue happens when **vanilla** store is used with **persist** middleware and inside state we have **boolean** values. At `createStore` call typescript throws an error `Type 'boolean' is not assignable to type 'false'` when we provide initial state. If value is any other type than boolean- no error happens
![image](https://github.com/pmndrs/zustand/assets/16834152/089b028c-8612-4471-ad1f-d2252821d1e9)

## Link to reproduction
https://codesandbox.io/s/laughing-mclaren-84v8tw?file=/src/index.ts

**Comment by De-Santa** [comment]
Think I found workaround

![image](https://github.com/pmndrs/zustand/assets/16834152/634b0999-22eb-4a56-8d69-d75bcd3785d8)

https://codesandbox.io/s/practical-sea-gp3gsh

**Comment by dai-shi** [comment]
Not sure if this is fixable at the library end. @devanshj has any idea?

I would probably do `false as boolean` workaround.

**Reply by devanshj** [reply, marked as answer]
I haven't taken a closer look but it seems to me that it's a TypeScript issue and not a zustand issue. As you too said the workaround is to write `foo: true as boolean`

**Reply by justynspooner** [reply]
I've hit a similar issue using Zustand 4.4.3 and typescript 5.2.2

I have strict mode on but discovered if I turn off `strictFunctionTypes` then the error goes away.

```
"strict": true,
"strictFunctionTypes": false
```

```
import { create } from 'zustand';
import { persist } from 'zustand/middleware';

interface AuthState {
  isAuthenticated: boolean;
}

export default create<AuthState>()(
  persist(
    () => ({
      isAuthenticated: false,
    }),
    {
      name: 'authStore',
    },
  ),
);

```

**Reply by dai-shi** [reply]
Yeah, but that workaround sounds worse than `as boolean`.

**Comment by eurobob** [comment]
@dai-shi I'm getting this issue even with the provided example in the documentation, so something is not correct

<img width="503" alt="Screenshot 2023-11-01 at 05 53 05" src="https://github.com/pmndrs/zustand/assets/4255350/2e39269d-a5f1-44a0-a3ee-39d1cfe1142e">
<img width="503" alt="Screenshot 2023-11-01 at 05 52 56" src="https://github.com/pmndrs/zustand/assets/4255350/cfb28d4d-127e-4885-b863-c10005f4e1d7">

**Reply by eurobob** [reply]
Never mind, i see this is already reported https://github.com/pmndrs/zustand/discussions/2163
